<a href="https://colab.research.google.com/github/blakedshull/datascience/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

I am sorry, I accidentally uploaded the wrong file.  This was my exploration and the data sets will be on the github link.

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np

# My first choice is this data on credit card defaults

In [ ]:
filename = '/content/default of credit card clients.csv'
df = pd.read_csv(filename)
new_header = df.iloc[0] 
df = df[1:] 
df.columns = new_header
df.head()

0 ID LIMIT_BAL SEX EDUCATION MARRIAGE AGE PAY_0 PAY_2 PAY_3 PAY_4  ...  \
1  1     20000   2         2        1  24     2     2    -1    -1  ...   
2  2    120000   2         2        2  26    -1     2     0     0  ...   
3  3     90000   2         2        2  34     0     0     0     0  ...   
4  4     50000   2         2        1  37     0     0     0     0  ...   
5  5     50000   1         2        1  57    -1     0    -1     0  ...   

0 BILL_AMT4 BILL_AMT5 BILL_AMT6 PAY_AMT1 PAY_AMT2 PAY_AMT3 PAY_AMT4 PAY_AMT5  \
1         0         0         0        0      689        0        0        0   
2      3272      3455      3261        0     1000     1000     1000        0   
3     14331     14948     15549     1518     1500     1000     1000     1000   
4     28314     28959     29547     2000     2019     1200     1100     1069   
5     20940     19146     19131     2000    36681    10000     9000      689   

0 PAY_AMT6 default payment next month  
1        0                          1  
2     2000                          1  
3     5000                          0  
4     1000                          0  
5      679                          0  

[5 rows x 25 columns]

In [ ]:
df.shape

(30000, 25)

### 1) Source of data

I found this data on UCI's machine learning repository.  It is data on credit card defaults, also in Taiwan.

### 2) Brief description of data
There are 30,000 data points, with a wide variety of data including their card limit, gender, education, marriage, age, education (1 = graduate school; 2 = university; 3 = high school; 4 = others), Marital status (1 = married; 2 = single; 3 = others), history of payments (X6 = the repayment status in September, 2005; X7 = the repayment status in August, 2005; . . .;X11 = the repayment status in April, 2005. The measurement scale for the repayment status is: -1 = pay duly; 1 = payment delay for one month; 2 = payment delay for two months, etc)
X12-X17: Amount of bill statement (NT dollar). X12 = amount of bill statement in September, 2005; X13 = amount of bill statement in August, 2005; . . .; X17 = amount of bill statement in April, 2005.
X18-X23: Amount of previous payment (NT dollar). X18 = amount paid in September, 2005; X19 = amount paid in August, 2005; . . .;X23 = amount paid in April, 2005. 

### 3) What is the target?
The target is the default payment next month (1 = yes 0 = no)

### 4) Is this a classification or regression problem?
This is a classification problem, people fit into yes or no.

### 5) How many features?
25

### 6) How many rows of data.
There are 30,000 rows of data.

### 7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
One challenge in this data set is not organized perfectly, I will probably have to rework history of payments, amount of bill statement per month, and amount of prev payment.

# My second choice is Taiwanese Real Estate data.

In [ ]:
filename2 = '/content/Real estate valuation data set.xlsx'
df2 = pd.read_excel(filename2)
df2.head()

No  X1 transaction date  X2 house age  \
0   1          2012.916667          32.0   
1   2          2012.916667          19.5   
2   3          2013.583333          13.3   
3   4          2013.500000          13.3   
4   5          2012.833333           5.0   

   X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                84.87882                               10   
1                               306.59470                                9   
2                               561.98450                                5   
3                               561.98450                                5   
4                               390.56840                                5   

   X5 latitude  X6 longitude  Y house price of unit area  
0     24.98298     121.54024                        37.9  
1     24.98034     121.53951                        42.2  
2     24.98746     121.54391                        47.3  
3     24.98746     121.54391                        54.8  
4     24.97937     121.54245                        43.1

In [ ]:
df2.shape

(414, 8)

### 1) Source of data

I found this data on UCI's machine learning repository.  It is gathered from Taiwan.

### 2) Brief description of data
This table contains  414 houses, sale date, age of house, distance to mass rapid transit station, number of conveience stores within walking distance, latitude, longitude, and price per ping (1 ping = 3.3m squared)
### 3) What is the target?
The target is the house price of unit area.  

### 4) Is this a classification or regression problem?
This is a regression problem, price is a numerical set of values.

### 5) How many features?
8, although 1 of them is just ID and is therefore useless.

### 6) How many rows of data.
There are 414 rows of data.

### 7) What, if any, challenges do your foresee in cleaning, exploring, or modeling with this dataset?
One challenge in this data set is the limited area, this is only one district in one city.  

This data is also old, it may not be as useful.